### PR TITLE
controller: Fix domain migration

### LINKED ISSUE #2987: migrate-domain - fails with Git: Unable to read current working directory
I tried to use my own domain for a simplistic one node test cluster in AWS. The flynn cluster was just setup without any apps installed. Latest stable flynn version `v20160624.1`

I have added two DNS records
`A: cloud.mycompany.com ---> [IP OF FLYNN HOST]`
`CNAME: *.cloud.mycompany.com --> cloud.mycompany.com`

After testing that cloud.mycompany.com really resolves to the flynn host IP, I have started the migration:

`flynn cluster migrate-domain cloud.mycompany.com`

```
Migrate cluster domain from "abcd.flynnhub.com" to "cloud.mycompany.com"? (yes/no): yes
Migrating cluster domain (this can take up to 2m0s)...
Changed cluster domain from "abcd.flynnhub.com" to "cloud.mycompany.com"
error "exit status 128" running "git config --global http.https://git.cloud.mycompany.com.sslCAInfo /Users/isnull/.flynn/ca-certs/flynn-1466842130.pem": "fatal: Unable to read current working directory: No such file or directory\n"
```
The new domain has been properly added to flynn cli and set to default. However, this left the flynn cli in a bad state, since now it has the wrong certifcate/pin for this cluster:

`flynn apps` for example yields now:

`Get https://controller.cloud.mycompany.com/apps: pinned: the peer leaf certificate did not match the provided pin`

PS: I reported this issue just to help you fix the migrate-domain, I already deleted this flynn cluster.

### PR DESCRIPTION
**Pull request by jvatic**
Closes #2987

### CODE REVIEW

File: controller/worker/domain_migration/domain_migration.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func (m *migration) appMaybeCreateRoute(appID string, oldRoute *router.Route, ro
 		Sticky:  oldRoute.Sticky,
 		Service: oldRoute.Service,
 	}
-	if oldRoute.Certificate != nil && oldRoute.Certificate.Cert == m.dm.OldTLSCert.Cert {
+	if oldRoute.Certificate != nil && oldRoute.Certificate.Cert == strings.TrimSuffix(m.dm.OldTLSCert.Cert, "\n") {
```
Comment (titanous) [inline controller/worker/domain_migration/domain_migration.go]: Should this use `TrimSpace`?

Comment (jvatic) [inline controller/worker/domain_migration/domain_migration.go]: Yeah, that would be safer. It should only ever have a trailing newline at most though.

**Comment by jvatic** [comment]
@titanous comment addressed

**Comment by titanous** [comment]
LGTM

**Pull request merged**